src/useWrappedRefWithWarning.js

```diff
@@ -0,0 +1,23 @@
+import invariant from 'invariant';
+import { useCallback } from 'react';
+import useMergedRefs from '@restart/hooks/useMergedRefs';
+
+const isReactClassComponent = c => c && c.isReactComponent;
+
+export default function useWrappedRefWithWarning(ref, componentName) {
+  if (!__DEV__) return ref;
+
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const warningRef = useCallback(
+    refValue => {
+      invariant(
+        refValue == null || !isReactClassComponent(refValue),
+        `${componentName} injected a ref to a provided \`as\` component that resolved to a component instance instead of a DOM element. ` +
+          'Use `React.forwardRef` to provide the injected ref to the class component as a prop in order to pass it directly to a DOM element',
+      );
+    },
+    [componentName],
+  );
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  return useMergedRefs(warningRef, ref);
```

Comment:
I don't understand the value of the wrap/merge approach here. Why not just print out a warning?

Reply:
The warning needs to happen in the ref handler itself, not in the hook call. It's effectively a callback ref that doesn't do anything with the value.

Reply:
okay, i see... why can't we just use `findDOMNode` here?

Reply:
We could but I'm attempting to remove the need here, to close out strict mode warnings

Reply:
What if we fall back to using `findDOMNode` if the ref is a react component? Then maybe emit some appropriate warning?

Reply:
idk that seems unnecessary, since this is a deprecated pattern we should probably use the moment for making breaking changes to switch over. It's an easy enough workaround in user code if folks want the old behavior